### PR TITLE
fix(kprompt): proceed on enter

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Allow trigger proceed action on _Enter_ upon typing in confirmation text in KPrompt

### CODE REVIEW

File: docs/components/prompt.md [hunk 1]
```diff
@@ -86,7 +86,7 @@ A string to be displayed as the prompt dialog title. Can also be [slotted](#titl
 
 ### confirmationText
 
-A string the user must type before the action button becomes enabled.
+A string the user must type before the action button becomes enabled. Pressing _Enter_ after typing in the confirmation text will trigger proceed action too.
```
Comment (adamdehaven) [inline docs/components/prompt.md:89]:
```suggestion
A string the user must type before the action button becomes enabled. Pressing `Enter` in the confirmation text input will trigger the `proceed` action.
```

File: src/components/KPrompt/KPrompt.vue [hunk 2]
```diff
@@ -139,6 +146,7 @@ const sanitizedAttrs = computed(() => {
 })
 
 const confirmationInput = ref<string>('')
+const displayErrorState = ref<boolean>(false)
```
Comment (adamdehaven) [inline src/components/KPrompt/KPrompt.vue:149]: image

issue: if the error state is triggered, and then I close the prompt and re-open it, the error state persists. I believe this ref should be reset when closing/canceling

File: src/components/KPrompt/KPrompt.vue [hunk 3]
```diff
@@ -112,11 +115,15 @@ const props = defineProps({
     type: Object as PropType<ModalAttributes>,
     default: () => ({}),
   },
+  errorMessage: {
+    type: String,
+    default: 'Please enter the text as indicated above.',
```
Comment (adamdehaven) [inline src/components/KPrompt/KPrompt.vue:120]:
```suggestion
    default: 'You must enter the text to proceed.',
```

Comment (portikM) [reply src/components/KPrompt/KPrompt.vue:120]: It's not just enter the text, it also has to match.. so how about _You must enter the text as indicated above to proceed._?

File: src/components/KPrompt/KPrompt.vue (hunk 2 above)
Comment (portikM) [reply src/components/KPrompt/KPrompt.vue:149]: Done

File: src/components/KPrompt/KPrompt.vue (hunk 3 above)
Comment (adamdehaven) [reply src/components/KPrompt/KPrompt.vue:120]: sgtm, except what about `"to confirm" instead of "proceed"?

Comment (portikM) [reply src/components/KPrompt/KPrompt.vue]: Done

**Review by adamdehaven — APPROVED** [review]
lgtm

**Pull request merged**